### DISCUSSION TITLE
Vogen + Result Types

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [SteveDunn/Vogen#649] [Q&A] [answered]
I'm working on converting a custom type of mine that represents a United States Social Security Number to use Vogen.

One of the features I would like to preserve with this type is being able to return a `Result<SocialSecurityNumber>`, wherein the `Result` type contains a the value or an error. The intent and purpose behidn the `Result` type is to avoid throwing exceptions.

At the moment, I have this:

```cs
public static Result<SocialSecurityNumber> TryParseResult(string ssn)
{
    Validation validationResult = Validate(NormalizeInput(ssn));

    return validationResult == Validation.Ok
        ? From(ssn) // double validation
        : new ValidationError(validationResult.Message);
```

This code has not been tested, but should allow me to effectively replicate what `From(string)` does in this situation while avoiding throwing. I would simply new up an instance of my type here but doing so is done in error, so I'm stuck with double validating the string.

Is there a better way of handling this situation?

**Comment by SteveDunn** [comment, marked as answer]
Hi @Foxtrek64 - thanks for the question. Yes, there is a better way that avoids exceptions. You can use the `TryParse` methods.
One returns a `ValueObjectOrError` and one return a `bool` and takes an `out` parameter.
I've just [documented these here](https://stevedunn.github.io/Vogen/create-and-use-value-objects.html). Please feel free to edit that page if anything could be improved (there's an 'Edit page' link at the top).
I've also added some [unit tests ](https://github.com/SteveDunn/Vogen/blob/main/tests/ConsumerTests/TryFromTests.cs) here that demonstrate its use.

I hope that answers your question.

**Reply by TheBrambleShark** [reply]
Thanks for the quick response! This is exactly what I'm looking for.
